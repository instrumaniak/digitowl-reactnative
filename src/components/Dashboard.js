import React from 'react'

import {
  View,
  ScrollView,
  StyleSheet
} from 'react-native'

import {
  Card,
  CardItem,
  Text,
  Button
} from 'native-base'


import { LineChart, Grid, XAxis, YAxis } from 'react-native-svg-charts'
import { Circle } from 'react-native-svg'
import { colors } from '../config'
import CircleButton from './CircleButton'


const data = [ 50, 10, 40, 95, -4, -24, 85, 91, 35, 53, -53, 24, 50, -20, -80 ]

const Decorator = ({ x, y, data }) => {
  return data.map(( value, index) => (
    <Circle
      key={index}
      cx={ x(index) }
      cy={ y(value)}
      r={ 4 }
      stroke={ colors.grey }
      strokeWidth={3}
      fill={colors.whiteSmoke}
    />
  ))
}

const Dashboard = () => (
  <View style={styles.container}>
    <ScrollView>

      {/* Wallets & Account */}
      <Card style={{ elevation: 2 }}>
        <CardItem>
          <Text style={styles.cardHeaderText}>Wallets & Accounts</Text>
        </CardItem>
        <CardItem>
          <View style={styles.btnContainer}>
            <Button style={styles.btnLight} light>
              <Text style={styles.btnText} uppercase={false}>Add Cash Wallet </Text>
            </Button>
            <Button style={styles.btnLight} light>
              <Text style={styles.btnText} uppercase={false}>Connect Bank </Text>
            </Button>
          </View>
        </CardItem>
      </Card>

      {/* Overview */}
      <Card style={{ elevation: 2 }}>
        <CardItem>
          <Text style={styles.cardHeaderText}>Overview</Text>
        </CardItem>
        <CardItem>
          <View style={styles.btnContainer}>
            <View style={[styles.overviewButton, styles.oBtnActive]}>
              <Text style={styles.obtnHeading}>Total Wealth</Text>
              <Text style={styles.obtnValue}>+34 628 USD</Text>
            </View>
            <View style={styles.overviewButton}>
              <Text style={styles.obtnHeading}>Monthly Cash Flow</Text>
              <Text style={styles.obtnValue}>+257 USD</Text>
            </View>
          </View>
        </CardItem>

        {/* Charts */}
        <CardItem>
          <View style={{ flexDirection: 'row'}}>
            <YAxis
              data={data}
              style={{ marginBottom: 30, marginRight: 10 }}
              contentInset={{ top: 10, bottom: 10 }}
              svg={{ fill: colors.grey1 }}
            />
            <View style={{ flex: 1}}>
              <LineChart
                data={data}
                style={{ height: 200 }}
                svg={{stroke: colors.grey1, strokeOpacity: 0.6, strokeWidth: 3 }}
                contentInset={{ top: 10, bottom: 10 }}
              >
                <Decorator />
                <Grid svg={{strokeOpacity: 1, stroke: colors.grey}}/>
              </LineChart>
              <XAxis
                style={{ marginHorizontal: -10, height: 30}}
                data={data}
                contentInset={{ left: 10, right: 10 }}
                formatLabel={(value, index) => index}
                svg={{ fill: colors.grey1 }}
              />
            </View>
          </View>
        </CardItem>
      </Card>
    </ScrollView>
    <CircleButton />
  </View>
)

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 5,
    flex: 1
  },
  cardHeaderText: {
    color: colors.grey1
  },
  btnContainer: {
    flex: 1,
    flexDirection: 'row'
  },
  btnLight: {
    backgroundColor: 'white',
    flex: 1,
    marginHorizontal: 5,
    justifyContent: 'center'
  },
  btnText: {
    color: colors.greenDark
  },
  overviewButton: {
    flex: 1,
    alignItems: 'center',
    padding: 5
  },
  oBtnActive: {
    backgroundColor: 'rgba(0,0,0,0.05)',
    borderRadius: 5
  },
  obtnHeading: {
    color: colors.grey1,
    marginBottom: 5,
    fontSize: 18
  },
  obtnValue: {
    color: colors.grey2,
    fontSize: 25
  }
})

export default Dashboard
