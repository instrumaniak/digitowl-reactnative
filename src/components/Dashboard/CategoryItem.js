import React from 'react'
import PropTypes from 'prop-types'
import {
  View,
  Text,
  StyleSheet
} from 'react-native'

import { colors } from '../../config'

const CategoryItem = ({
  iconText,
  title,
  description,
  value
}) => (
  <View style={styles.container}>
    <View style={styles.iconContainer}>
      <Text style={styles.iconText}>{iconText}</Text>
    </View>
    <View style={styles.dataContainer}>
      <View>
        <Text style={styles.title}>{ title }</Text>
        <Text style={styles.description}>{ description }</Text>
      </View>
      <View>
        <Text style={styles.value}>{ value }</Text>
      </View>
    </View>
  </View>
)

CategoryItem.propTypes = {
  iconText: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string,
  value: PropTypes.string
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
  },
  iconContainer: {
    width: 50,
    height: 50,
    backgroundColor: colors.grey,
    borderRadius: 25,
    justifyContent: 'center',
    alignItems: 'center'
  },
  iconText: {
    color: colors.white,
    //fontWeight: 'bold',
    fontSize: 35
  },
  dataContainer: {
    flex: 1,
    paddingLeft: 15,
    flexDirection: 'row',
    justifyContent: 'space-between'
  },

  // typography
  title: {
    color: colors.grey1,
    fontSize: 20,
    //fontWeight: 'bold'
  },
  description: {
    color: colors.grey,
    fontSize: 16
  },
  value: {
    color: colors.grey1,
    fontSize: 18,
    fontWeight: 'bold'
  }
})

export default CategoryItem
