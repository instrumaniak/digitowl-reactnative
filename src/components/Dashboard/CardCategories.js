/**
 *   Dashboard: Categories section
 */

import React, { PureComponent } from 'react'
import { View, StyleSheet } from 'react-native'
import { PieChart } from 'react-native-svg-charts'
import { Circle, G, Line, Text as SText } from 'react-native-svg'

import {
  Card,
  CardItem,
  Text
} from 'native-base'

import { human } from 'react-native-typography'
import { colors } from '../../config'
import CategoryItem from './CategoryItem'

const data2 = [ 50, 53, 24, 5, 20, 80 ]

const pieData = data2
  .filter(value => value > 0)
  .map((value, index) => ({
    value,
    svg: {
      fill: colors.grey1
    },
    key: `pie-${index}`
  }))

const PieLabels = ({ slices }) => {
  return slices.map((slice, index) => {
    const { labelCentroid, pieCentroid, data } = slice
    return (
      <G key={index}>
        <Line
          x1={ labelCentroid[0] }
          y1={ labelCentroid[1] }
          x2={ pieCentroid[0] }
          y2={ pieCentroid[1] }
          stroke={ data.svg.fill }
        />
        <Circle
          cx={ labelCentroid[0] }
          cy={ labelCentroid[1] }
          r={ 20 }
          fill={ colors.grey }
        />
        <SText
          x={ labelCentroid[0] - 7 }
          y={ labelCentroid[1] + 5 }
          style={{ ...human.caption2Object }}
          stroke={ colors.grey2 }
        >
          { data.value} %
        </SText>
      </G>
    )
  })
}

class CardCategories extends PureComponent {
  render() {
    return (
      <Card style={{ elevation: 2 }}>
        <CardItem>
          <Text style={styles.cardHeaderText}>Categories</Text>
        </CardItem>
        <CardItem>
          <View style={styles.btnContainer}>
            <View style={[styles.overviewButton, styles.oBtnActive]}>
              <Text style={styles.obtnHeading}>Total Expenses</Text>
              <Text style={styles.obtnValue}>- 4 628 USD</Text>
            </View>
            <View style={styles.overviewButton}>
              <Text style={styles.obtnHeading}>Total Income</Text>
              <Text style={styles.obtnValue}>+ 5 884 USD</Text>
            </View>
          </View>
        </CardItem>

      {/* Chart */}
        <CardItem>
          <PieChart
            style={{ flex:1, height: 250 }}
            data={pieData}
            innerRadius={30}
            outerRadius={60}
            labelRadius={90}
          >
            <PieLabels />
          </PieChart>
        </CardItem>

      {/* Category Item list */}
        <CardItem>
          <CategoryItem
            iconText="$"
            title="Food and Drink"
            description="26 transations"
            value="-1 804.90 USD"
          />
        </CardItem>
        <CardItem>
          <CategoryItem
            iconText="$"
            title="Travel"
            description="17 transactions"
            value="-694.20 USD"
          />
        </CardItem>
        <CardItem>
          <CategoryItem
            iconText="$"
            title="Rent"
            description="1 transactions"
            value="-555.40 USD"
          />
        </CardItem>
        <CardItem>
          <CategoryItem
            iconText="$"
            title="Entertainment"
            description="5 transations"
            value="-462.80 USD"
          />
        </CardItem>
        <CardItem>
          <CategoryItem
            iconText="$"
            title="Energy"
            description="6 transactions"
            value="-450.50 USD"
          />
        </CardItem>

        <CardItem>
          <View style={{
            flex: 1,
            alignItems: 'center'
          }}>
            <Text style={{
              ...human.subheadObject,
              color: colors.grey1,
              textDecorationLine: 'underline'
            }}>All Categories (7)</Text>
          </View>
        </CardItem>
      </Card>
    )
  }
}

const styles = StyleSheet.create({
  cardHeaderText: {
    ...human.headlineObject,
    color: colors.grey2
  },
  btnContainer: {
    flex: 1,
    flexDirection: 'row'
  },
  btnLight: {
    backgroundColor: 'white',
    flex: 1,
    marginHorizontal: 5,
    justifyContent: 'center'
  },
  overviewButton: {
    flex: 1,
    alignItems: 'center',
    padding: 5
  },
  oBtnActive: {
    backgroundColor: 'rgba(0,0,0,0.05)',
    borderRadius: 5
  },
  obtnHeading: {
    ...human.bodyObject,
    color: colors.grey1,
    marginBottom: 5
  },
  obtnValue: {
    ...human.title3Object,
    color: colors.grey2
  }
})

export default CardCategories
