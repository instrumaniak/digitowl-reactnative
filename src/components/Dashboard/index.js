import React from 'react'

import {
  View,
  ScrollView,
  StyleSheet
} from 'react-native'

import {
  Card,
  CardItem,
  Text,
  Button
} from 'native-base'

import { human } from 'react-native-typography'
import { colors } from '../../config'

import CircleButton from '../CircleButton'
import CardOverview from './CardOverview'
import CardCategories from './CardCategories'
import CardHashtags from './CardHashtags'
import CardPlaces from './CardPlaces'

const Dashboard = () => (
  <View style={styles.container}>
    <ScrollView>

      {/* Wallets & Account */}
      <Card style={{ elevation: 2 }}>
        <CardItem>
          <Text style={styles.cardHeaderText}>Wallets & Accounts</Text>
        </CardItem>
        <CardItem>
          <View style={styles.btnContainer}>
            <Button style={styles.btnLight} light>
              <Text style={styles.btnText} uppercase={false}>Add Cash Wallet </Text>
            </Button>
            <Button style={styles.btnLight} light>
              <Text style={styles.btnText} uppercase={false}>Connect Bank </Text>
            </Button>
          </View>
        </CardItem>
      </Card>

      <CardOverview />

      <Card style={{ elevation: 2 }}>
        <CardItem>
          <View style={styles.btnContainer}>
            <View style={styles.overviewButton}>
              <Text style={styles.obtnHeading}>Average Daily Expense</Text>
              <Text style={styles.obtnValue}>- 28.47 USD</Text>
            </View>
            <View style={styles.overviewButton}>
              <Text style={styles.obtnHeading}>Last Month Comparison</Text>
              <Text style={styles.obtnValue}>+22%</Text>
            </View>
          </View>
        </CardItem>
      </Card>

      <CardCategories />

      <Card style={{ elevation: 2 }}>
        <CardItem>
          <Text style={styles.cardHeaderText}>Transfers</Text>
        </CardItem>
        <CardItem>
          <View style={styles.btnContainer}>
            <View style={styles.overviewButton}>
              <Text style={styles.obtnHeading}>Outgoing</Text>
              <Text style={styles.obtnValue}>- 125 USD</Text>
            </View>
            <View style={styles.overviewButton}>
              <Text style={styles.obtnHeading}>Incoming</Text>
              <Text style={styles.obtnValue}>+1 200 USD</Text>
            </View>
          </View>
        </CardItem>
      </Card>

      <CardHashtags />
      <CardPlaces />
    </ScrollView>
    <CircleButton />
  </View>
)

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 5,
    flex: 1
  },
  cardHeaderText: {
    ...human.headlineObject,
    color: colors.grey2
  },
  btnContainer: {
    flex: 1,
    flexDirection: 'row'
  },
  btnLight: {
    backgroundColor: 'white',
    flex: 1,
    marginHorizontal: 5,
    justifyContent: 'center'
  },
  btnText: {
    ...human.bodyObject,
    color: colors.grey1
  },
  overviewButton: {
    flex: 1,
    alignItems: 'center',
    padding: 5
  },
  oBtnActive: {
    backgroundColor: 'rgba(0,0,0,0.05)',
    borderRadius: 5
  },
  obtnHeading: {
    ...human.subheadObject,
    color: colors.grey1,
    marginBottom: 5
  },
  obtnValue: {
    ...human.title3Object,
    color: colors.grey2
  }
})

export default Dashboard
