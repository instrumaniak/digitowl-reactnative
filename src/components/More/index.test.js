import React from 'react'
import Dashboard from './'
import renderer from 'react-test-renderer'

test('Renders correctly', () => {
  const tree = renderer.create(<Dashboard />).toJSON()
  expect(tree).toMatchSnapshot()
})
