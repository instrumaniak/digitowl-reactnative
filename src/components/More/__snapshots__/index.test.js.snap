// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 5,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderLeftWidth": 0.5,
            "borderRadius": 2,
            "borderRightWidth": 0.5,
            "borderTopWidth": 0.5,
            "elevation": 3,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            General
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#5B6267",
                      "fontSize": 18,
                    }
                  }
                >
                  John Doe
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#c3c6c7",
                      "fontSize": 16,
                    }
                  }
                >
                  user.email@host.net
                </Text>
              </View>
              <View>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "color": "#000",
                          "fontSize": 28,
                          "width": 35,
                        },
                        Object {
                          "color": "#5B6267",
                        },
                      ],
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#5B6267",
                      "fontSize": 18,
                    }
                  }
                >
                  Upgrade Options
                </Text>
              </View>
              <View>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "color": "#000",
                          "fontSize": 28,
                          "width": 35,
                        },
                        Object {
                          "color": "#5B6267",
                        },
                      ],
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#5B6267",
                      "fontSize": 18,
                    }
                  }
                >
                  Notifications
                </Text>
              </View>
              <View>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "color": "#000",
                          "fontSize": 28,
                          "width": 35,
                        },
                        Object {
                          "color": "#5B6267",
                        },
                      ],
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderLeftWidth": 0.5,
            "borderRadius": 2,
            "borderRightWidth": 0.5,
            "borderTopWidth": 0.5,
            "elevation": 3,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Wallet & Accounts
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#5B6267",
                      "fontSize": 18,
                    }
                  }
                >
                  Manual Wallets
                </Text>
              </View>
              <View>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "color": "#000",
                          "fontSize": 28,
                          "width": 35,
                        },
                        Object {
                          "color": "#5B6267",
                        },
                      ],
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#5B6267",
                      "fontSize": 18,
                    }
                  }
                >
                  Bank accounts
                </Text>
              </View>
              <View>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "color": "#000",
                          "fontSize": 28,
                          "width": 35,
                        },
                        Object {
                          "color": "#5B6267",
                        },
                      ],
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#5B6267",
                      "fontSize": 18,
                    }
                  }
                >
                  All Categories
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#c3c6c7",
                      "fontSize": 16,
                    }
                  }
                >
                  17 Expenses, 8 Income
                </Text>
              </View>
              <View>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "color": "#000",
                          "fontSize": 28,
                          "width": 35,
                        },
                        Object {
                          "color": "#5B6267",
                        },
                      ],
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#5B6267",
                      "fontSize": 18,
                    }
                  }
                >
                  Main Currency
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#c3c6c7",
                      "fontSize": 16,
                    }
                  }
                >
                  United States Dollars(USD)
                </Text>
              </View>
              <View>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "color": "#000",
                          "fontSize": 28,
                          "width": 35,
                        },
                        Object {
                          "color": "#5B6267",
                        },
                      ],
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderLeftWidth": 0.5,
            "borderRadius": 2,
            "borderRightWidth": 0.5,
            "borderTopWidth": 0.5,
            "elevation": 3,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Other
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#5B6267",
                      "fontSize": 18,
                    }
                  }
                >
                  Advanced
                </Text>
              </View>
              <View>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "color": "#000",
                          "fontSize": 28,
                          "width": 35,
                        },
                        Object {
                          "color": "#5B6267",
                        },
                      ],
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#5B6267",
                      "fontSize": 18,
                    }
                  }
                >
                  Support
                </Text>
              </View>
              <View>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "color": "#000",
                          "fontSize": 28,
                          "width": 35,
                        },
                        Object {
                          "color": "#5B6267",
                        },
                      ],
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#5B6267",
                      "fontSize": 18,
                    }
                  }
                >
                  Terms & Policies
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#c3c6c7",
                      "fontSize": 16,
                    }
                  }
                >
                  17 Expenses, 8 Income
                </Text>
              </View>
              <View>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "color": "#000",
                          "fontSize": 28,
                          "width": 35,
                        },
                        Object {
                          "color": "#5B6267",
                        },
                      ],
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 5,
            "paddingVertical": 10,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#f4f4f4",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16.5,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                Object {
                  "color": "#CC2936",
                },
              ]
            }
            uppercase={false}
          >
            Log Out
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
